community/values/src/main/java/org/neo4j/values/storable/DurationValue.java

```diff
@@ -192,10 +192,20 @@ DurationValue create(
 
     private static final DurationValue ZERO = new DurationValue( 0, 0, 0, 0 );
     private static final List<TemporalUnit> UNITS = unmodifiableList( asList( MONTHS, DAYS, SECONDS, NANOS ) );
+    // This comparator is safe until 292,271,023,045 years. After that, we have an overflow.
```

Comment:
This makes it safe for all "natural" durations - i.e. the ones that can be obtained by asking for the duration between two instants that can be represented. Probably good enough, although I think we can do even better if we wanted to.

Reply:
That would require going to BigInteger, though.

Reply:
Only if we want it to pass through as a single object somewhere, which isn't necessary.